File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.MessagingService.Models/inotification_item.ts



import { NotificationRecipient_Item } from './notification-recipient_item';

export interface INotification_Item {
  
	NotificationID: number;
	NotificationSubject: string;
	NotificationMessage: string;
	EmailMessage: string;
	Unread: boolean;
	NotificationContextTypeID: number;
	NotificationContextType: string;
	ContextID: number;
	ModifiedDate: Date;
	NotificationMessageID: number;
	ModifiedByAppUserID: number;
	ModifiedByAppUser: string;
	MessageTemplateName: string;
	IncludeContextLink: boolean;
	AppUserID: number;
	ViewedDate?: Date;
	EmailSentDate?: Date;
	NotificationRecipientModifiedDate: Date;
	Recipients?: NotificationRecipient_Item[];

}

File: GTTS/INL.MessagingService/Models/INotification_Item.cs
using System;
using System.Collections.Generic;

namespace INL.MessagingService.Models
{
    public interface INotification_Item
    {
        long NotificationID { get; set; }
        string NotificationSubject { get; set; }
        string NotificationMessage { get; set; }
        string EmailMessage { get; set; }
        bool Unread { get; set; }
        int NotificationContextTypeID { get; set; }
        string NotificationContextType { get; set; }
        long ContextID { get; set; }
        DateTime ModifiedDate { get; set; }
        int NotificationMessageID { get; set; }
        int ModifiedByAppUserID { get; set; }
        string ModifiedByAppUser { get; set; }
        string MessageTemplateName { get; set; }
        bool IncludeContextLink { get; set; }
        int AppUserID { get; set; }
        DateTime? ViewedDate { get; set; }
        DateTime? EmailSentDate { get; set; }
        DateTime NotificationRecipientModifiedDate { get; set; }

        List<NotificationRecipient_Item> Recipients { get; set; }
    }
}
